pkg/test/framework/components/istio/operator.go

```diff
@@ -128,62 +134,146 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
+		return fmt.Errorf("failed to write iop: %v", err)
 	}
+
 	cmd := []string{
 		"manifest", "apply",
 		"--skip-confirmation",
 		"--logtostderr",
 		"-f", iopFile,
-		"--set", "values.global.imagePullPolicy=" + s.PullPolicy,
 		"--wait",
 	}
+
+	if c.environment.IsMulticluster() {
+		// Set the clusterName for the local cluster.
+		// This MUST match the clusterName in the remote secret for this cluster.
+		cmd = append(cmd, setArg("values.global.multiCluster.clusterName", clusterName(cluster))...)
+	}
+
 	// If control plane values set, assume this includes the full set of values, and .Values is
 	// just for helm use case. Otherwise, include all values.
 	if cfg.ControlPlaneValues == "" {
 		for k, v := range cfg.Values {
-			cmd = append(cmd, "--set", fmt.Sprintf("values.%s=%s", k, v))
+			// Add the "values." prefix if not specified.
+			if !strings.HasPrefix(k, "values.") {
+				k = "values." + k
+			}
+			cmd = append(cmd, setArg(k, v)...)
 		}
 	}
 
-	scopes.CI.Infof("Running istioctl %v", cmd)
+	scopes.CI.Infof("Running istio control plane on cluster %d %v", cluster.Index(), cmd)
 	if _, err := istioCtl.Invoke(cmd); err != nil {
-		return nil, fmt.Errorf("manifest apply failed: %v", err)
+		return fmt.Errorf("manifest apply failed: %v", err)
 	}
 
 	if !cfg.SkipWaitForValidationWebhook {
 		// Wait for the validation webhook to come online before continuing.
-		if _, _, err = env.KubeClusters[0].WaitUntilServiceEndpointsAreReady(cfg.SystemNamespace, "istio-pilot"); err != nil {
+		if _, _, err = cluster.WaitUntilServiceEndpointsAreReady(cfg.SystemNamespace, "istio-pilot"); err != nil {
 			err = fmt.Errorf("error waiting %s/%s service endpoints: %v", cfg.SystemNamespace, "istio-pilot", err)
 			scopes.CI.Info(err.Error())
-			i.Dump()
-			return nil, err
+			c.Dump()
+			return err
 		}
 
 		// Wait for webhook to come online. The only reliable way to do that is to see if we can submit invalid config.
-		err = waitForValidationWebhook(env.KubeClusters[0].Accessor, cfg)
+		err = waitForValidationWebhook(cluster.Accessor, cfg)
 		if err != nil {
-			i.Dump()
-			return nil, err
+			c.Dump()
+			return err
 		}
 	}
+	return nil
+}
 
-	return i, nil
+func setArg(key, value string) []string {
+	return []string{"--set", key + "=" + value}
+}
+
+func createRemoteSecret(ctx resource.Context, cluster kube.Cluster) (string, error) {
+	istioCtl, err := istioctl.New(ctx, istioctl.Config{
+		Cluster: cluster,
+	})
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{
+		"x", "create-remote-secret",
+		"--name", clusterName(cluster),
+	}
+
+	scopes.CI.Infof("Creating remote secret for cluster cluster %d %v", cluster.Index(), cmd)
+	if out, err := istioCtl.Invoke(cmd); err != nil {
+		return "", fmt.Errorf("create remote secret failed for cluster %d: %v", cluster.Index(), err)
+	} else {
+		return out, nil
+	}
 }
+
+func clusterName(cluster kube.Cluster) string {
+	return fmt.Sprintf("cluster-%d", cluster.Index())
```

Comment:
Should clusterName be a method on `kube.Cluster`?

Reply:
done.